### PR TITLE
add per-key RGB for Quark LP RGB variants

### PR DESCRIPTION
**Pull request by npspears**
add support for Quark LP MX RGB and Quark LP Choc RGB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/checkerboards/quark_lp/config.h [hunk 1]
```diff
@@ -46,19 +46,11 @@
...
-#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
-#define RGBLIGHT_EFFECT_SNAKE
-#define RGBLIGHT_EFFECT_KNIGHT
-#define RGBLIGHT_EFFECT_CHRISTMAS
-#define RGBLIGHT_EFFECT_STATIC_GRADIENT
-#define RGBLIGHT_EFFECT_RGB_TEST
-#define RGBLIGHT_EFFECT_ALTERNATING
-#define RGBLIGHT_EFFECT_TWINKLE
-#define RGBLED_NUM 14 // number of LEDs
+#define RGBLIGHT_ANIMATIONS // run RGB animations
```
Comment (drashna) [inline keyboards/checkerboards/quark_lp/config.h:51]: please revert this change. 

We want to explicitly enable modes rather than use a blanket enable.

Comment (npspears) [reply keyboards/checkerboards/quark_lp/config.h]: understood and done

Comment (drashna) [reply keyboards/checkerboards/quark_lp/config.h]: thanks!

**Comment by npspears** [comment]
> Also, if this is actually per key RGB, the rgb matrix feature would be a much better option over RGB light (underglow), as the animations and such are actually designed for per key rgb.

Unfortunately, I don't think I have time or the understanding (currently) to implement it as a matrix. Perhaps in the future I or someone else can revisit this. If acceptable, let's move forward with the current implementation.

**Comment by drashna** [comment]
The biggest thing is that you need: 
https://docs.qmk.fm/#/feature_rgb_matrix?id=common-configuration

**Comment by npspears** [comment]
ok, I was able to find some time to learn how to write the code. Please help as this is my first time

File: keyboards/checkerboards/quark_lp/config.h [hunk 2]
```diff
@@ -46,19 +46,67 @@
 /* Locking resynchronize hack */
 #define LOCKING_RESYNC_ENABLE
 
-// ws2812 options
+// RGB Matrix
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/checkerboards/quark_lp/config.h:50]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/checkerboards/quark_lp/rules.mk [hunk 3]
```diff
@@ -16,3 +16,4 @@ NKRO_ENABLE = no            # Enable N-Key Rollover
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```
Comment (drashna) [inline keyboards/checkerboards/quark_lp/rules.mk:17]: since it looks like the two features share the led strip... probably want to disable rgb light. 
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by npspears** [comment]
Thanks Drashna!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]
